packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/LegacyManifest.java

```diff
@@ -171,29 +170,15 @@ private Uri getAssetsUrlBase() {
 
       // assetUrlOverride may be an absolute or relative URL
       // if relative, we should resolve with respect to the manifest URL
+      // java.net.URI's resolve method does this for us
       String assetsPathOrUrl = manifestJson.optString("assetUrlOverride", "assets");
-      Uri maybeAssetsUrl = Uri.parse(assetsPathOrUrl);
-      if (maybeAssetsUrl != null && maybeAssetsUrl.isAbsolute()) {
-        return maybeAssetsUrl;
-      } else {
-        String normalizedAssetsPath;
-        try {
-          URI assetsPathURI = new URI(assetsPathOrUrl);
-          normalizedAssetsPath = assetsPathURI.normalize().toString();
```

Comment:
Add a test case that tests the normalization behavior, perhaps with an asset path like `./a/../b`. URI.resolve normalizes so the tests should pass. (Also I'm curious what happens if the asset path has enough `..`s that it goes back past the hostname.)